### PR TITLE
Zephyr getting started guide

### PR DESCRIPTION
**Pull request by noahp**

### CODE REVIEW

**Comment by noahp** [comment]
a bit sparse, but a starting point!

File: docs/mcu/zephyr-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+```ini
+CONFIG_CONSOLE=y
+
+# this board has a uart console
+CONFIG_UART_CONSOLE=y
+CONFIG_SHELL=y
+
+# enable logs as well
+CONFIG_LOG=y
+```
```
Comment (noahp) [inline docs/mcu/zephyr-guide.mdx:104]: Todo add the shell commands and memfault upload reference stuff

File: docs/mcu/zephyr-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+Depending on your project setup, the `west.yml` file may be in the application
+repository (most common), or in a separate manifest repository. Read more about
+supported development repositories here:
+
+- <https://docs.zephyrproject.org/latest/develop/west/workspaces.html#topologies-supported>
+
+## Enabling Memfault
+
+To enable Memfault, there's a few required Kconfig options that should be set:
```
Comment (tyhoff) [inline docs/mcu/zephyr-guide.mdx:62]:
```suggestion
To enable Memfault, there are a few required Kconfig options that should be set:
```

File: docs/mcu/zephyr-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+   If these files are not provided by the user, they will be silently ignored.
+   See the Kconfig options for some control over how these files are included.
+
+## Zephyr `__ASSERT()`
+
+Zephyr RTOS has a built-in runtime assert system, see more information here:
+
+- <https://docs.zephyrproject.org/3.1.0/kernel/services/other/fatal.html>
+
+Memfault coredump capture will by default be triggered when the system
```
Comment (tyhoff) [inline docs/mcu/zephyr-guide.mdx:205]: Something like:

"The Memfault coredump capture routine will by default be triggered ..."

or

"Memfault's coredump capture routine"

**Review by tyhoff — APPROVED** [review]

**Review by victorbordo — APPROVED** [review]

**Pull request merged**